de.dlr.sc.virsat.uiengine.ui/src/de/dlr/sc/virsat/uiengine/ui/editor/snippets/general/UiSnippetRoleManagement.java

```diff
@@ -109,6 +109,23 @@ private void openCustomDialog() {
 			customDialog.open();
 		}
 	}
+	/**
+	 * Method to check if the user has the required permission
+	 */
+	
+	private boolean hasPermission() {
+	    String currentUser = UserRegistry.getInstance().getUserName();
+
+	    // Check if the current user is in the list of users for the selected discipline
+	    Discipline selectedDiscipline = getSelectedDiscipline();

```

Comment:
Hi Dimitri, this code is also implemented in the RightsHelper... I think it's better to delegate the RightsHelper (RightHelper.hasWritePermission) rather than reimplementing it... Otherwise we have to duplicate code in multiple locations..